geode-core/src/main/java/org/apache/geode/internal/cache/EntryEventImpl.java

```diff
@@ -2871,11 +2906,15 @@ public void release() {
    * Return true if this EntryEvent may have off-heap references.
    */
   private boolean mayHaveOffHeapReferences() {
+    if (this.offHeapLock == null) {
+      return false;
+    }
+
     InternalRegion lr = getRegion();
     if (lr != null) {
       return lr.getOffHeap();
     }
-    // if region field is null it is possible that we have off-heap values
```

Comment:
I think you should add this comment back. It explains why, in this case, we return true instead of false